### ISSUE TITLE
Parsing and type checking with FSharp.Compiler.Service?

### ISSUE THREAD

**Issue by dsyme** [penberg/fjord#6]
Hi @penberg 

Have you thought about using [FSharp.Compiler.Service](http://fsharp.github.io/FSharp.Compiler.Service/) and generating JVM code from [the resolved expression trees](http://fsharp.github.io/FSharp.Compiler.Service/typedtree.html) and [symbols](http://fsharp.github.io/FSharp.Compiler.Service/symbols.html) that component gives you?

If you succeeded, you could then bootstrap FSharp.Compiler.Service to be a completely JVM component.

At least, I'd encourage you to take a look at that. I think that's where I'd start if getting F# on the JVM.

Cheers!
@dsyme

**Comment by Neftedollar** [comment]
@dsyme awesome thing!

**Comment by alexpantyukhin** [comment]
Yes, great thing! I think the main goal which could be achieved there it to try compile FSharp.Compiler.Service to JVM. If this aim will be achieved then It would be possible to make java based F# compiler.

**Comment by Neftedollar** [comment]
@alexpantyukhin yeah, like fable
